internal/core/saptune/saptune.go

```diff
@@ -0,0 +1,120 @@
+package saptune
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/utils"
+	"golang.org/x/mod/semver"
+)
+
+var (
+	ErrSaptuneVersionUnknown = errors.New("could not determine saptune version")
+	ErrUnsupportedSaptuneVer = errors.New("saptune version is not supported")
+)
+
+const (
+	MinimalSaptuneVersion = "v3.1.0"
+)
+
+type Saptune struct {
+	Version  *string         `json:"version"`
+	Commands []SaptuneOutput `json:"commands"`
+}
+type SaptuneOutput struct {
```

Comment:
Do we need all this structs here?
I mean, we are just bypassing the json, we don't use them at all internally

Reply:
No need and can be removed. I thought they might be used in the future if we need to do something with the JSON output before sending it, but I can remove & re-add when needed.